### ISSUE TITLE
Multilingual Speech Translation with Efficient Finetuning of Pretrained Models

### ISSUE THREAD

**Issue by devgunn** [devgunn/paper-review#1]
[Multilingual Speech Translation with Efficient Finetuning of Pretrained Models](https://arxiv.org/abs/2010.12829)

We present a simple yet effective approach to build multilingual speech-to-text (ST) translation by efficient transfer learning from pretrained speech encoder and text decoder. Our key finding is that a minimalistic LNA (LayerNorm and Attention) finetuning can achieve zero-shot crosslingual and cross-modality transfer ability by only finetuning less than 10% of the pretrained parameters. This enables effectively leveraging large pretrained models with low training cost. Using wav2vec 2.0 for acoustic modeling, and mBART for multilingual text generation, our approach advanced the new state-of-the-art for 34 translation directions (and surpassing cascaded ST for 23 of them) on large-scale multilingual ST benchmark CoVoST 2 (+6.4 BLEU on average across 15 En-X directions and +5.1 BLEU on average across 19 X-En directions). Our approach demonstrates strong zero-shot performance in a many-to-many multilingual model (+5.7 BLEU on average across 18 non-English directions), making it an appealing approach for attaining high-quality speech translation with improved parameter and data efficiency.

**Comment by devgunn** [comment]
- 문제제기
  - Speech-to-text 번역에서 다국어 번역을 수행할 때, Pretrained model 을 활용한 더 효과적인 방법을 제시
  - One-model-for-all 패러다임을 확장

- 개선점
  - Speech-to-text 번역을 수행하기 위해, Pre-trained Single Modality(Speech/Text)  모델들을 효과적이고 간단한 방법으로 결합
    - 최소한의 구조 변경과 cross-modal adaptor 를 추가함 (audio 인코더의 output과 text 디코더의 input 의 길이 차이 극복을 위해)
    - Multi-task(Speech-to-text 와 Text-to-text) fine-tuning 을 통해서 성능 향상
  - Layer Norm 과 Attention 만을 fine-tuning 하는 효율적인 transfer learning 이 가능
    - 학습 하지 못한 언어 방향 처리를 위한 zero-shot cross-lingual transfer 에 효과적(A->B training 후 A->C 그리고 C->B 에 test)
    - Zero-shot Multilingual 번역에 효과적 (A->B , B->C training 후, A->C 에 test)
  - Pre-trained Audio Encoder(wav2vec) 과 Multilingual Text Decoder(mBART) 를 활용하려, CoVoST 2 와 Europarl 에 대한 SOTA 획득

- 방법론
  - Pre-trained modules
    - Wav2vec 2.0
      - Unlabelled 오디오 데이터로부터, High quality speech representation 을 학습
      - Feature Encoder : convolutional layer 를 통과시켜서 latent speech representation 을 학습, latent speech representation 은 transformer 의 입력으로 주어짐 
      - Context Encoder : Transformer 에 입력된 latent speech representation 으로 부터 context representation 를 학습(span masking 을 통해)
      - Train 방법은 contrastive task 로 distractor 로 부터 true latent 를 구분하는 방법
    - mBART
      - 디노이징 오토 인코더 모델이며, 랜덤 마스킹/토큰 순서 변경 형태로 노이지 인풋을 만들어 원형을 맞추는 형태로 학습
      - N개 언어의 모노 데이터로 학습
  - Length Adaptor
    - Encoder 와 Decoder 사이에 Light weight adaptor 모듈을 추가
    - Adaptor 는 audio 와 text 사이의 길이 차이를 해결하기 위해 projection 과 downsampling 을 수행
    - 음성 sequence 를 2^n 으로 줄이기 위해 stack of 1-dim convolutional layer with stride 2 수행
  - LNA Finetuning
    - Parameter 들을 효율적으로 fine-tuning 하기 위해 Layer Normalization 과 Multi-head Attention 만 fine-tune
    - LNA 의 목적은 pretraining 단계와 downstream(ST) task 의 차이를 최소화 하는 것이 목적
    - 그렇게 해야하는 근거는 아래의 2가지 (*그러나 이부분이 이해가 잘 안가는 부분)
      - Layer Norm
        - Layer Norm 부분의 parameter 들은 pretrain 때 사용한 데이터 분포에 맞춰져 있기 때문에 down stream task 에 맞춰서 fine-tuning 이 필요 (근데 안그런 부분이 있는지?, motivation 이 잘 이해가 안됨)
      - Attention
        - Pre-trained 디코더의 Encoder Attention(Encoder-decoder attention 을 말하는 듯) text-to-text MT task
  - Join Speech-to-text Fine-tuning
    - Pretrained 된 모델에 MT 와 ST task 를 이용해서 jointly fine-tuning 을 수행
    - Speech-to-text 데이터 셋에서, {Transcript : Transcript}, {Transcript : Translation} 을 각각 MT 와 ST 의 데이터로 활용

- 훈련법
  - Encoder : wav2vec 2.0 large 를 이용하여 intialize, encoder output 은 convolution 을 통과시켜 down-sampling 수행
  - Decoder : mBART 를 이용
    - 50 mono 언어로 pre-train -> 49개 언어의 bi-texts 로 train
    - X-En Speech Translation 은 MBART-ML50N1(49 lang -> En) 이용
    - En-X Speech Translation 은 MBART-ML501N(En -> 49 lang) 이용
  - LNA finetuning 
    - Multilingual 으로 오는 성능 boosting 없이, 순전히 bi-lingual 세팅만으로 LNA finetuning 의 1) parameter efficiency 2) crosslingual transfer ability 를 확인
    - 추가적으로 LNA-E(encoder only), LNA-D(decoder only), LAN-E,D(both) 에 대해서도 evaluate
  - Joint Tranining
    - 2개의 Encoder 는 input 이 text 경우 mBART, speech 경우 wav2vec 으로 intialized 되었는데, wav2vec Encoder 의 12개 transformer layer 는 mBART 인코더 layer 로 교체됨
      - _The last 12 transformer layers in the wav2vec encoder is replaced with 12 mBART encoder layers._ (무슨말인지)
    - 그리고 이 2개의 인코더는 joint training 동안 shared 됨
    - Decoder 또한 2개의 task 에서 shared 되고, pretrained mBART decoder 에서 intialize 됨.
    - MT task 훈련을 위해서 추가적인 ML50 bitext 를 사용
    - mBART 인코더를 speech encoder 로 사용해서, pre-trained 모델의 big mismatch 가 발생하기 때문에 전체 parameter 모두를 fine-tune
  -Baselines
    - From scratch : seq-to-seq Transformer w/o any pretraining
    - ASRPT+Multi : Pretraining Encoder on ASR and ST (결과 비교에서는 Multilingual ASR pretrain -> Multilingual ST 모델을 사용)
    - XMEF(Cross-Modal Efficient Finetuning)-BL : XMEF, LNA 기법으로 Bilingual fientuning 비교


- 결과
  - Parameter Efficiency
    - Bilingual ST(En-De, De-En in CoVoST) 로 evaluate
    - LNA 의 경우 LNA-Minimalist (69M params) 사용
    - <img width="351" alt="image" src="https://user-images.githubusercontent.com/13184386/127637625-2a3843de-3079-45f6-b66e-68c9d280b1fc.png">
  - Crosslingual Transfer
    - 영어 로만 학습된 wav2vec 인코더의 다른 언어에 대한 transfer ability 는, CoVoST 2 De-En ST task 로 검증
    - LNA-ESA(Encoder Self-Attention) 으로 비교
    - Finetuning entiere encoder(All) vs Finetuning feature extractor(Input)  
    - 전체 비교군 LNA-ESA, LNA-Min, All, Input
    - <img width="354" alt="image" src="https://user-images.githubusercontent.com/13184386/127638315-315f970f-e36f-40ca-a54f-996b479bb2fe.png">
  - Multilingual Finetuning
    - Multilingual Finetuning 을 통해서 cross-lingual transfer 능력을 evaluation
    - Source-side(Speech) Transfer
      - 목적 및 방법
        - 본 논문의 기법이 crosslingual transfer (unseen language 로 부터 번역하는) 에 동작하는지 확인
        - 5 to-English 언어 set 으로 fine-tune 하고, unseen language(Pt) 의 ST 능력을 확인
      - 발견
        - 더 많은 parameter 를 fine-tuning 하는 LNA-D, All, LNA-(E,D) 의 경우 2배 더 빠르게 훈련되고, seen & unseen language 에 대한 더 나은 generalization 능력을 가짐
    - Target-side(Text) Transfer
      - 목적 및 방법
        - 본 논문 기법이 완전히 새로운 언어로의 번역 즉, zero-shot transfer capability 능력을 보여줌.
      - 발견
        - unseen language(Ja) 로의 task 는 LNA-D 가 zero-shot transfer 에 큰 영향을 끼침
        - Decoder 전체를 finetuning 하는 것은 seen language 를 더 잘 optimize 하는 것을 발견
  - Multilingual Speech Translation
    - XMEF(Cross-Modal Efficient Finetuning) 를 Multilingual finetuning 한 결과를 비교
    - CoVoST 2의 36개 translation direction(X-En :21, En-X:15) 에 대해서 evaluate
      - Many-to-one(X-En) 각각
        - LNA-E,D(전체 파라미터의 20%만 사용) 를 했을때 Finetune All 보다 성능 향상(1.3 BLEU)
        - 이후 LNA-D 를 했을때 추가적인 성능향상도 있엇음
      - One-to-many(En-X) 각각
        - LNA-E 의 경우 Finetune All 과 거의 유사한 성능이 나왔으며, Finetune All 보다 40% 적은 파라미터로 훈련.
        - LNA-E,D 의 경우 Pretrained model 에 비해 8~20% 정도만의 파라미터로 훈련하되, 성능은 강력한 baseline (ASR PT + Multilingual Finetuning) 과 유사하게 유지
        - 단 mBART 가 보지못했던 Ca, Cy 언어에 대해서는 효과를 보지 못했음
      - Joint Training
        - Many-to-one(X-En)
          - Speech 훈련 데이터의 양이 18시간 정도가 넘고, 병렬 코퍼스가 1M 이 넘는 경우에는 single task 에 비해서 성능이 우수하게 나옴
          - 그러나 speech 훈련 데이터의 양이 너무 적으면(10시간 미만), joint training 은 효과적이지 않았고 심지어 성능 저하 시키기도 함.
        - One-to-many(En-X)
          - En 음성 훈련 데이터 양이 364시간으로 훈련했을 때, joint training 잘 동작
  - Zero-shot
    - Zero-shot multilingual translation(Train on X-> En, En->Y, test on X->Y)
      - Europarl의 En-X 와 X-En 훈련 데이터를 갖고 LNA-D 를 적용하여 multilingual finetuning 을 하였을 때, to-En 과 from-En 에 대해서, supervised performance 와 병렬 코퍼스 없는 zero-shot performance 가 strong baseline(parallel data 를 이용한 many-to-many multilingual model) 보다 높은 수준으로 성능 향상이 있었음.
